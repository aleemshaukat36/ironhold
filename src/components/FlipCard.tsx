import { ReactNode } from "react";

export type FlipCardProps = {
  frontContent: ReactNode;
  backContent: ReactNode;
  visibleSide: String | "front" | "back";
};

export default function FlipCard({
  frontContent,
  backContent,
  visibleSide,
}: FlipCardProps) {
  return (
    <div className="flip-card">
      <div
        className="flip-card-inner"
        style={visibleSide === "back" ? { transform: "rotateY(180deg)" } : {}}
      >
        <div className="flip-card-front">{frontContent}</div>
        <div className="flip-card-back">{backContent}</div>
      </div>
    </div>
  );
}
