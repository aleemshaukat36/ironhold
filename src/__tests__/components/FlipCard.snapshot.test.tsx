import { render } from "@testing-library/react";
import FlipCard from "../../components/FlipCard";

it("renders homepage unchanged", () => {
  const { container } = render(
    <FlipCard frontContent={<></>} backContent={<></>} visibleSide={"front"} />
  );
  expect(container).toMatchSnapshot();
});

it("renders homepage unchanged", () => {
  const { container } = render(
    <FlipCard frontContent={<></>} backContent={<></>} visibleSide={"back"} />
  );
  expect(container).toMatchSnapshot();
});
