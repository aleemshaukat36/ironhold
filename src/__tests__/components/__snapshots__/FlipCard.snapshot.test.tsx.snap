// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="flip-card"
  >
    <div
      class="flip-card-inner"
    >
      <div
        class="flip-card-front"
      />
      <div
        class="flip-card-back"
      />
    </div>
  </div>
</div>
`;

exports[`renders homepage unchanged 2`] = `
<div>
  <div
    class="flip-card"
  >
    <div
      class="flip-card-inner"
      style="transform: rotateY(180deg);"
    >
      <div
        class="flip-card-front"
      />
      <div
        class="flip-card-back"
      />
    </div>
  </div>
</div>
`;
